ebos/equil/initstateequil.hh

```diff
@@ -1773,6 +1770,28 @@ private:
     Vec rs_;
     Vec rv_;
     Vec swatInit_;
+    Vec cellCenterDepth_;
+    std::vector<std::pair<double,double>> cellZSpan_;
+    std::vector<std::pair<double,double>> cellZMinMax_;
+
+    void updateCellProps_(const GridView& gridView)
+    {
+        ElementMapper elemMapper(gridView, Dune::mcmgElementLayout());
+        int numElements = gridView.size(/*codim=*/0);
+        cellCenterDepth_.resize(numElements);
+        cellZSpan_.resize(numElements);
+        cellZMinMax_.resize(numElements);
+
+        auto elemIt = gridView.template begin</*codim=*/0>();
+        const auto& elemEndIt = gridView.template end</*codim=*/0>();
+        for (; elemIt != elemEndIt; ++elemIt) {
+            const Element& element = *elemIt;
+            const unsigned int elemIdx = elemMapper.index(element);
+            cellCenterDepth_[elemIdx] = Details::cellCenterDepth(element);
```

Comment:
One question here, why not use the `cellCenterDepth_` from the `vanguard` here? Thanks. @totto82

Reply:
Never mind. I thought without this can simply some work for me. It is not case.